rust/flatbuffers/src/push.rs

```diff
@@ -24,7 +24,11 @@ use crate::endian_scalar::emplace_scalar;
 /// types.
 pub trait Push: Sized {
     type Output;
-    fn push(&self, dst: &mut [u8], _rest: &[u8]);
+
+    /// SAFETY
+    ///
+    /// dst is aligned to [`Self::alignment`] and has length greater than or equal to [`Self::size`]
+    unsafe fn push(&self, dst: &mut [u8], written_len: usize);
```

Comment:
It is worth highlighting the change here to not pass the `_rest` slice. Without this LLVM struggles to optimise push correctly as there is no stable unchecked version of `split_at_mut` and the presence of the conditional check causes loops to not vectorize correctly

Reply:
Can you share the difference in benchmarks that this makes?

Reply:
Without any changes, create_byte_vector_100_naive would manage 200MB/s. With some tweaking to reserve space up-front, etc... this got up to 2000MB/s, with this change it manages 8333MB/s